src/main/java/thewaypointers/trafficsimulator/TrafficSimulatorManager.java

```diff
@@ -11,15 +11,19 @@
     final static long TIME_STEP = 100;
     final static long SIMULATION_TIME_STEP = 300;
     static WorldStateDTO worldState;
+    static MainFrame mainFrame;
 
     public static void main(String[] args) {
 
         Simulation simulation = new Simulation();
-        MainFrame mainFrame = new MainFrame(worldState);
+
         while(true){
             try{
                 Thread.sleep(TIME_STEP);
                 worldState = simulation.getNextSimulationStep(SIMULATION_TIME_STEP);
+                if(mainFrame == null){
```

Comment:
i was getting an error if I do :
MainFrame.mapContainerPanel.mapPanel.NewStateReceived(worldState);
before the simulation actually creates the worldState object... so I made it like this so it does that only once not each time it enters the while loop (and it's after the first step of simulation so worldstate is not null)